### DISCUSSION TITLE
resetTiles not working

### DISCUSSION THREAD

**Discussion by bmason8** [fleaflet/flutter_map#1083] [Q&A]
I'm having an issue with successfully clearing map tiles when switch tile sets (default to satellite and vice versa) and wondering if anyone can help me out.

I've been using flutter_map for a year now along with map_map_tile_caching and it's been working great when I would just cache one type of tile (let's call them default) using `StorageCachingTileProvider.` Then when I would need to show satellite tiles, I would switch `NonCachingNetworkTileProvider` and all 'default' tiles would get cleared without issue. I did it this way because the library only supported caching one style of tile. *This still works fyi (ie no lack of tile clearing issues).

However, I've been wanting to cache two different styles of map (default and satellite) and now that `flutter_map_tile_caching 4.0.0-dev.7` is available I implemented that functionality. It works great however, when I switch map styles the previous map style remain in certain areas. I initially assumed it was a `flutter_map_tile_caching` issue but was told to take a look at this thread [(fleaflet/flutter_map#667)](https://github.com/fleaflet/flutter_map/pull/667). So I implemented the resetTiles strategy demonstrated in the [example lib](https://github.com/fleaflet/flutter_map/blob/master/example/lib/pages/reset_tile_layer.dart) but it's not working for me. I'm wondering if I've simply implemented it incorrectly or if it's a potential bug.

Here is how I'm implementing the tileProvider using mapbox (to be clear, everything else works great):
*also sorry for the formatting, I'm pulling my hair out trying to fix it but have no idea why it looks like it does.

MAP_VIEW PAGE:

`FlutterMap(
            mapController: _mapControlNotifier.mapController,
            options: _mapControlNotifier.mapOptions(context),
            children: [
              TileLayerWidget(options: _mapControlNotifier.getCachingTileLayerOptions()),
            ],
          ),`

MAP_CONTROL_NOTIFIER CLASS
`TileLayerOptions getCachingTileLayerOptions() {

    if (setupInProgress) {
      tileProvider = NonCachingNetworkTileProvider();
    } else {
      tileProvider = StorageCachingTileProvider(
        cachedValidDuration: Duration(days: 60),
        parentDirectory: mapTilesParentDirectory,
        storeName: storeName,
      );
    }
    return TileLayerOptions(
        reset: resetController.stream,
        tileProvider: tileProvider,
        urlTemplate: mapboxUrlTemplate,
        additionalOptions: {'accessToken': mapboxApiToken, 'id': (layerToggle) ? MAP_STYLE_ID_OUTDOORS : MAP_STYLE_ID_SATELLITE},
        // overrideTilesWhenUrlChanges: true,
        // tileFadeInDuration: 0,
        // tileFadeInStartWhenOverride: 1.0,
        backgroundColor: Colors.blueGrey);
  }`

I set up the stream controller a little differently because I'm using a ChangeNotifier.
`late StreamController<Null> resetController;`
Then in a setup() function that gets run before the map tiles are loaded I initialize it.
`resetController = StreamController.broadcast();`
And then I run _resetTiles() when the user presses the 'Swap tile layers' button.
`void _resetTiles() {
layerToggle = !layerToggle;
notifyListeners();
resetController.add(null);
  }`

I've also tried a few different things to see if I'm doing something else wrong and none of them made any difference.

1. Added the TileLayerOptions via layers[] instead of children.
2. Used two separate TileLayerOptions for each map style.
3.  Used a different tile provider for the second map style (mapbox and openstreetmap).
4. Yelling at my computer.

Any help would be greatly appreciated. I've been in contact with the dev from the `flutter_map_tile_caching` package and although it seems like it's a mistake I'm making or potentially a `flutter_map` bug, he'll be taking another look at it sometime soon when he does some more testing.
I'm really hoping it's just a stupid mistake I'm making but I haven't figured it out yet so please let me know if you have any ideas for me to try. 
Thanks.

**Comment by JaffaKetchup** [comment]
Plugin author here. For discussion thread, please see toward the bottom of https://github.com/JaffaKetchup/flutter_map_tile_caching/discussions/29#discussioncomment-1607415.

@bmason8 Out of interest, now I know you're using provider, that might be causing an issue. When you reset, does a blank frame flash? If it doesn't, it doesn't look like the call is being made.
Can you try setting up your provider like in the 'state/generalProvider.dart' in the latest commit to the v4 branch - https://github.com/JaffaKetchup/flutter_map_tile_caching/blob/v4/example/lib/state/general_provider.dart? Then you don't need to initialize it, and you should just be able to call `providerRef.resetMap()` when you change the store?
This probably won't work, but i think it's worth a try. Let me know if it doesn't work, or  if the blank frame flash does happen at the moment :).

**Reply by JaffaKetchup** [reply]
@bmason8 Hi there,
I've done some more testing, and I can now see it's not working as it should even with `resetTiles()` in use. I have a feeling this is not an issue with `flutter_map`, so is it OK if we return to the previous discussion thread?
I am working on it now, to see if I can fix it.
Thanks!

**Reply by bmason8** [reply]
@JaffaKetchup sorry I didn't get notified that you replied to this thread and just happened to check it today! Absolutely, I'll jump back to the previous thread to continue this. I just tried what you recommended above and it didn't work either. A blank frame does flash though for what it's worth.
